### PR TITLE
[1880] add stub fix + typos

### LINKED ISSUE #8753: [1880] Beijing exit spaces not protected
It is possible to lay track which blocks a not yet started company from building out of Beijing. This should not be allowed.
[hs_ffzpfsdg_1674941485.json.txt](https://github.com/tobymao/18xx/files/10528373/hs_ffzpfsdg_1674941485.json.txt)

### PR DESCRIPTION
**Pull request by bentziaxl**
add stubs are restricted (fix #8753 )
add terminal to offboards so FI connection doesn't wrongly close FIs
fix typos 
fix SR log showing SR 5.1 when it should be SR 5

### CODE REVIEW

File: lib/engine/game/g_1880/game.rb [hunk 1]
```diff
@@ -712,7 +713,7 @@ def lessee
         def round_description(name, round_number = nil)
           round_number ||= @round.round_num
           description = super
-          description += ".#{round_number}" if @round.is_a?(Engine::Round::Operating)
+          description += ".#{round_number}" if name == self.class::OPERATING_ROUND_NAME
           description += " - Train Marker at #{@train_marker.name}" if @train_marker && !@end_game_triggered
```
Comment (bentziaxl) [inline lib/engine/game/g_1880/game.rb:717]: needed change, timing issue.
When round_description is called for logging during an SR, it's still the OR, so the previous conditional will resolve to true even when i want to log the SR round. 
https://github.com/tobymao/18xx/blob/master/lib/engine/game/base.rb#L2627

File: lib/engine/game/g_1880/game.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@ class Game < Game::Base
         include_meta(G1880::Meta)
         include Map
         include Entities
+        include StubsAreRestricted
```
Comment (ollybh) [inline lib/engine/game/g_1880/game.rb:16]: Do you also need to `require_relative '../stubs_are_restricted'` before including this module?

File: lib/engine/game/g_1880/step/check_fi_connection.rb [hunk 3]
```diff
@@ -87,12 +87,26 @@ def process_percent(action)
 
           def process_token(action)
             fi = @merging[:fi]
+<<<<<<< HEAD
```
Comment (ollybh) [inline lib/engine/game/g_1880/step/check_fi_connection.rb]: This merge conflict needs to be resolved.

File: lib/engine/game/g_1880/game.rb (hunk 2 above)
Comment (bentziaxl) [reply lib/engine/game/g_1880/game.rb:17]: yep, added

File: lib/engine/game/g_1880/step/check_fi_connection.rb (hunk 3 above)
Comment (bentziaxl) [reply lib/engine/game/g_1880/step/check_fi_connection.rb]: i removed it....where are you seeing this?

Comment (ollybh) [reply lib/engine/game/g_1880/step/check_fi_connection.rb]: My bad, I'd clicked onto the wrong page and was looking at the first commit, not the full pull request.

**Review by crericha — APPROVED** [review]

**Pull request merged**